src/main/java/org/graylog/integrations/aws/inputs/AWSInput.java

```diff
@@ -0,0 +1,167 @@
+import org.graylog.integrations.aws.AWSUtils;
+import org.graylog.integrations.aws.codecs.AWSMetaCodec;
+import org.graylog.integrations.aws.service.AWSService;
+import org.graylog.integrations.aws.transports.KinesisTransport;
+import org.graylog2.plugin.LocalMetricRegistry;
+import org.graylog2.plugin.ServerStatus;
+import org.graylog2.plugin.buffers.InputBuffer;
+import org.graylog2.plugin.configuration.Configuration;
+import org.graylog2.plugin.configuration.ConfigurationRequest;
+import org.graylog2.plugin.configuration.fields.ConfigurationField;
+import org.graylog2.plugin.configuration.fields.DropdownField;
+import org.graylog2.plugin.configuration.fields.TextField;
+import org.graylog2.plugin.inputs.MessageInput;
+import org.graylog2.plugin.inputs.MisfireException;
+import org.graylog2.plugin.inputs.annotations.ConfigClass;
+import org.graylog2.plugin.inputs.annotations.FactoryClass;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.regions.Region;
+
+import javax.inject.Inject;
+
+/**
+ * General AWS input for all types of supported AWS logs.
+ */
+public class AWSInput extends MessageInput {
+
+    public static final String NAME = "AWS Input";
+    public static final String TYPE = "org.graylog.integrations.aws.inputs.AWSInput";
+
+    private static final Logger LOG = LoggerFactory.getLogger(AWSInput.class);
+
+    /**
+     * Specifies one of the {@code AWSInputType} choices, which indicates which codec and transport
+     * should be used.
+     */
+    public static final String CK_AWS_INPUT_TYPE = "aws_input_type";
+    public static final String CK_TITLE = "title";
+    public static final String CK_DESCRIPTION = "description";
+    public static final String CK_GLOBAL = "global";
+    public static final String CK_AWS_REGION = "aws_region";
+    public static final String CK_ACCESS_KEY = "aws_access_key";
+    public static final String CK_SECRET_KEY = "aws_secret_key";
+    public static final String CK_ASSUME_ROLE_ARN = "aws_assume_role_arn";
+
+    @Inject
+    public AWSInput(@Assisted Configuration configuration,
+                    MetricRegistry metricRegistry,
+                    KinesisTransport.Factory transport,
+                    LocalMetricRegistry localRegistry,
+                    AWSMetaCodec.Factory codec,
+                    Config config,
+                    Descriptor descriptor,
+                    ServerStatus serverStatus) {
+        super(metricRegistry,
+              configuration,
+              transport.create(configuration),
+              localRegistry,
+              codec.create(configuration),
+              config,
+              descriptor,
+              serverStatus);
+    }
+
+    @Override
+    public void launch(InputBuffer buffer) throws MisfireException {
+
+        LOG.info("Starting AWS Input...");
+        super.launch(buffer);
+    }
+
+    @Override
+    public void stop() {
+
+        LOG.info("Stopping AWS Input...");
```

Comment:
We already have logging statements for stopping inputs so I think this can go because it doesn't provide any further insights.